diag_utils/diag_utils/diagUtilsLib.py

```diff
@@ -95,8 +95,8 @@ def strip_prefix(indict, prefix):
 
     for k,v in indict.iteritems():
         if k.startswith(prefix):
-#            outdict[k[i:]] = v
-            outdict[k[8:]] = v
+            outdict[k[i:]] = v
+#            outdict[k[8:]] = v
```

Comment:
NOTE: the hard coding of eight character limit for stripping prefixes from env variables does not work with ilamb unless an non-intuitive prefix is used, e.g. replacing ILAMBDIAG with ILAMDIAG. Something simple like IMBDIAG also doesn't work because it has too-few characters.

But I'm not sure what the broader implications of changing this hard-coding back to the generic implementation is. It may break something in another component. Testing is needed before merging this PR!

Reply:
@bandre-ucar - this change broke the diags when testing by truncating too many
leading characters.

Reply:
I'm wondering if there is ambiguity about whether to include the '_' in the prefix, and therefore whether 'i = len(prefix)' or 'i = len(prefix) + 1'. The comments in the land diags seem to indicate that the '_' is included in the prefix, which means the i+1 is incorrect? 

Can you provide more information about what prefix is breaking?

Or maybe there is a simple grep command for something like "strip_prefix(envDict" that will show us all uses of strip_prefix...